pkg/couchdb/stream/all_docs.go

```diff
@@ -0,0 +1,350 @@
+// The stream package can be used for streaming CouchDB responses in JSON
+// format from the CouchDB cluster to a client, with the stack doing stuff like
+// filtering some fields. It is way faster that doing a full parsing of the
+// JSON response, doing stuff, and then reserialize to JSON for large payloads.
+package stream
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/ohler55/ojg/oj"
+)
+
+type allDocsFilter struct {
+	// config
+	fields   []string
+	skipDDoc bool
```

Comment:
I prefer to see documented fields rather than having to look for their usage to understand what they mean. But since they're private attributes maybe I'm looking at it the wrong way and I should focus on the public API.